src/domains/ValuesDomain.js

```diff
@@ -145,6 +145,13 @@ export default class ValuesDomain {
       return Add(realm, lnum, rnum);
     } else if (op === "<" || op === ">" || op === ">=" || op === "<=") {
       // ECMA262 12.10.3
+      let lprim = To.ToPrimitiveOrAbstract(realm, lval);
```

Comment:
Doing this here and then calling AbstractRelationalComparison below can cause side effects to happen more than once if lval has a side-effecting implicit conversion. This is probably why there are test262 failures.